akka-stream/src/main/scala/akka/stream/stage/GraphStage.scala

```diff
@@ -545,7 +545,7 @@ abstract class GraphStageLogic private[stream] (val inCount: Int, val outCount:
       // Detailed error information should not add overhead to the hot path
       ReactiveStreamsCompliance.requireNonNullElement(elem)
       require(!isClosed(out), s"Cannot push closed port ($out)")
-      require(isAvailable(out), s"Cannot push port ($out) twice")
+      require(isAvailable(out), s"Cannot push port ($out) twice or before being pulled")
```

Comment:
agree that it's a good improvement, perhaps `, or before it being pulled` wording is a bit more clear?